### ISSUE TITLE
Tracks not loading anymore

### ISSUE THREAD

**Issue by laurensbl** [eneiluj/gpxpod#21]
Hi,

After updating to 5.0.9 some directories with tracks aren't loading anymore. In the Nextcloud log is logged:

`Exception: OCA\GpxPod\Service\ProcessService::sanitizeGpxContent(): Return value must be of type string, null returned in file '/srv/www-local/nextcloud/apps/gpxpod/lib/Service/ProcessService.php' line 1174`

Fails to load in both new and old GUI. Another dir however, does load.

**Comment by julien-nc** [comment]
Hi, can you replace line 1174 of `gpxpod/lib/Service/ProcessService.php` with:
``` php
return preg_replace('/(<time>.*)\[[^]]*\](<\/time>)/', '$1$2', $content) ?? $content;
```
and try again?

**Comment by julien-nc** [comment]
I would be interested to see the track causing the preg_replace to return null. If one of the directories that does not load does not contain a lot of tracks, could you try to remove them one by one and try reloading the directory in GpxPod to identify a specific track that can't be loaded. I would be interested to get this track if it's not too personal.

**Comment by laurensbl** [comment]
> Hi, can you replace line 1174 of `gpxpod/lib/Service/ProcessService.php` with:
> 
> ```
> return preg_replace('/(<time>.*)\[[^]]*\](<\/time>)/', '$1$2', $content) ?? $content;
> ```
> 
> and try again?

After replacing the line 1174 all tracks / folders load again. Thanks!

> I would be interested to see the track causing the preg_replace to return null. If one of the directories that does not load does not contain a lot of tracks, could you try to remove them one by one and try reloading the directory in GpxPod to identify a specific track that can't be loaded. I would be interested to get this track if it's not too personal.

I've sent you an email with a link to the files.

**Comment by julien-nc** [comment]
Thanks for the feedback and for the tracks. So the problem is actually the `preg_replace` Php function having a strong limit on string size. Some of your files being big, not so big but big enough.

The real fix for this would be to split the string into chunks. For now the patch you tried is good enough. It's included in v5.0.10 which is out.

Feel free to reopen if needed.
